content/documentation/pages/6-recipes/4-kafka/1-ext-kafka-cluster-cf.md

```diff
@@ -0,0 +1,146 @@
+---
+path: 'recipes/kafka/ext-kafka-cluster-cf/'
+title: 'External Kafka Cluster'
+description: 'Connect to an external Kafka Cluster from Cloud Foundry'
+---
+
+# Connect to External Kafka Cluster
+
+The Pivotal Cloud Foundry does not have Apache Kafka as a managed service in the Marketplace.
+However, it is common for developers to develop and deploy applications to Cloud Foundry that interact with an external Kafka cluster.
+This recipe specifically walks through the developer expectations from Spring Cloud Stream, Spring Cloud Data Flow, and as well as the [Spring Cloud Data Flow for PCF](https://network.pivotal.io/products/p-dataflow) tile.
+
+We will review the required Spring Cloud Stream properties and how they are translated over to the applications for the following deployment options in Cloud Foundry.
+
+- Applications run as standalone app instances.
+- Applications deployed as part of a streaming data pipeline through open-source SCDF.
+- Applications deployed as part of a streaming data pipeline through [SCDF for PCF](https://network.pivotal.io/products/p-dataflow) tile.
+
+## Prerequisite
+
+Let's start with preparing the external Kafka cluster credentials.
+
+Typically, a series of Kafka brokers, collectively are referred to as a Kafka cluster.
+The brokers individually can be reached through its external IP addresses or a well-defined DNS route could be available to use as well.
+
+For this walk-through, though, we will stick to a simpler setup of 3-broker cluster with their DNS addresses being `foo0.broker.foo`, `foo1.broker.foo`, and `foo2.broker.foo`. The default port of the brokers is `9092`.
+
+If the cluster is secured, depending on the security option in use at the broker, different properties are expected to be supplied for when applications attempt to connect to the external cluster.
+Again, for simplicity, we will use `PlainLoginModule` with username as `test` and password as `bestest`.
+
+## User-provided Services vs. Spring Boot Properties
+
+The next question Cloud Foundry developers stumble upon is whether or not to set up Kafka connection as Cloud Foundry custom user-provided service (CUPS) or simply passing connection credentials as Spring Boot properties.
+
+In Cloud Foundry, there isn't a Spring Cloud Connector or CF JavaEnv support for Kafka, so by service-binding the Kafka CUPS with the application, you _will not_ automatically be able to parse `VCAP_SERVICES` and pass the connection credentials to the applications dynamically.
+Even with CUPS in place, it is your responsibility to parse the `VCAP_SERVICES` JSON and pass them as Boot properties, so no real benefit.
```

Comment:
Rather than say 'no real benefit' ... perhaps say 'this isn't as seamless as it is with RabbitMQ'.